Add PyPi publishing file (#2825)
openvinotoolkit/nncf: docs/PyPiPublishing.md

[![GitHub Release](https://img.shields.io/github/v/release/openvinotoolkit/nncf?color=green)](https://github.com/openvinotoolkit/nncf/releases)
[![Website](https://img.shields.io/website?up_color=blue&up_message=docs&url=https%3A%2F%2Fdocs.openvino.ai%2Flatest%2Fopenvino_docs_model_optimization_guide.html)](https://docs.openvino.ai/nncf)
[![Apache License Version 2.0](https://img.shields.io/badge/license-Apache_2.0-green.svg)](https://github.com/openvinotoolkit/nncf?tab=Apache-2.0-1-ov-file#readme)
[![PyPI Downloads](https://static.pepy.tech/badge/nncf)](https://pypi.org/project/nncf/)

# Neural Network Compression Framework (NNCF)

Neural Network Compression Framework (NNCF) provides a suite of post-training
and training-time algorithms for optimizing inference of neural networks in
[OpenVINO&trade;](https://docs.openvino.ai) with a minimal accuracy drop.

NNCF is designed to work with models from [PyTorch](https://pytorch.org/),
[TensorFlow](https://www.tensorflow.org/), [ONNX](https://onnx.ai/) and
[OpenVINO&trade;](https://docs.openvino.ai).

> The framework is organized as a Python package that can be built and used
> in a standalone mode. Its architecture is unified to make it easy to
> add different compression algorithms for both PyTorch and TensorFlow deep
> learning frameworks.

Author: ```suggestion
The framework is organized as a Python package that can be built and used
as a standalone tool. Its architecture is unified to make adding different
compression algorithms easy for both PyTorch and TensorFlow.
```

The framework is organized as a Python package that can be built and used
in a standalone mode. Its architecture is unified to make it easy to
add different compression algorithms for both PyTorch and TensorFlow deep
learning frameworks.

NNCF provides [samples](#demos-tutorials-and-samples) that demonstrate the
usage of compression algorithms for different use cases and models. See
compression results achievable with the NNCF-powered samples on the
[NNCF Model Zoo page](https://github.com/openvinotoolkit/nncf/blob/develop/docs/ModelZoo.md).

For more information about NNCF, see:
- [NNCF repository](https://github.com/openvinotoolkit/nncf)
- [user documentation](https://docs.openvino.ai/nncf)
- [NNCF API documentation](https://openvinotoolkit.github.io/nncf/autoapi/nncf/)
- [usage examples](https://github.com/openvinotoolkit/nncf/tree/develop/docs/usage)
- [Notebook tutorials](https://github.com/openvinotoolkit/openvino_notebooks/blob/latest/notebooks/README.md#model-training)

## Table of contents:

- [Key Features](#key-features)
- [Installation](#installation-guide)
- [Third-party integration](#third-party-repository-integration)

> - [NNCF Model Zoo](#nncf-compressed-model-zoo)

Reviewer: Those links would not work on PyPI because the `<a>` tag is absent in the corresponding paragraph.
Ref: https://github.com/pypa/readme_renderer/issues/169#issuecomment-808577486

Author: hopefully, they will now - added the tag

## Key Features

### Post-Training Compression Algorithms

| Compression algorithm                                                                                                                                       | OpenVINO  | PyTorch   | TensorFlow    | ONNX          |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------- | :-------: | :-------: | :-----------: | :-----------: |
| [Post-Training Quantization](https://github.com/openvinotoolkit/nncf/blob/develop/docs/usage/post_training_compression/post_training_quantization/Usage.md) | Supported | Supported | Supported     | Supported     |

> | [Weights Compression](https://github.com/openvinotoolkit/nncf/blob/develop/docs/usage/post_training_compression/weights_compression/Usage.md)               | Supported | Supported | Not supported | Not supported |

Reviewer 2: It looks like, key features is outdated relative to readme on develop. Please, update it.

- Exporting PyTorch compressed models to ONNX\* checkpoints and TensorFlow
  compressed models to SavedModel or Frozen Graph format, ready to use with
  [OpenVINO&trade; toolkit](https://docs.openvino.ai).
- Support for [Accuracy-Aware model training](https://github.com/openvinotoolkit/nncf/blob/develop/docs/usage/training_time_compression/other_algorithms/Usage.md#accuracy-aware-model-training)
  pipelines via the [Adaptive Compression Level Training](https://github.com/openvinotoolkit/nncf/blob/develop/docs/accuracy_aware_model_training/AdaptiveCompressionLevelTraining.md)
  and [Early Exit Training](https://github.com/openvinotoolkit/nncf/blob/develop/docs/accuracy_aware_model_training/EarlyExitTraining.md).

## Installation Guide

NNCF can be installed as a regular PyPI package:

```bash
pip install nncf
```

For detailed installation instructions, refer to the
[Installation](https://github.com/openvinotoolkit/nncf/blob/develop/docs/Installation.md) guide.

### System Requirements

- Ubuntu 18.04 or later (64-bit)
- Python 3.8 or later
- Supported frameworks:
  - PyTorch >=2.2, <2.4
  - TensorFlow >=2.8.4, <=2.15.1
  - ONNX ==1.16.0

>   - OpenVINO >=2022.3.0

Reviewer: @MaximProshin, we would need to support this and change it for every release along with the `Readme.md`.

Author: Is this information updated in the readme file for every release? We can either use this information or we can refer to the readme from the pypi page.

Reviewer 3: It's not updated every release, but yes, we will need to make sure it's up to date every time
